### ISSUE TITLE
CJS support

### ISSUE THREAD

**Issue by jmroon** [dfpc-coe/node-CoT#29]
Hi @ingalls,

We're looking at supporting some TAK use cases and came across this library. Kudos on the library. It's very well written and easy to follow, and seems like a solid foundation to get us started!

I went to test out a trial implementation, but unfortunately quickly hit a wall. We have a NestJS backend and are unfortunately stuck on CommonJS as a result. While I may normally use dynamic imports to get around this issue, the inclusion of ESM specific artifacts such as `import.meta` break this approach.

Now, I could fork the repo, or copy the existing code base, but I thought I'd reach out to see if you might be interested in adding CJS support to your publish step. I could certainly contribute to this effort, but this may require some modifications. I'm not terribly experienced in this particular area of library publishing, but I assume it would require removing ESM specific artifacts such as `import.meta`.

Would you be interested in my contribution here? Or would you rather we fork the repo for our own use case?

**Comment by ingalls** [comment]
Hey, glad you might find it useful, I'd be happy to answer questions and add docs as needed. So far it's only been used internally to power CloudTAK so I haven't spent a ton of time on external facing docs. I do however do my best to follow SemVer and not break things for folks

Yeah I took a quick look at dual publishing with something like `tsup` via

```
tsup index.ts --format cjs,esm
```

but ran into the same issue. If this is something you are interested in looking into and cutting a PR for I'd glady review and get it merged. We're in the middle of wildfire season so I'm a little pinched for time to dive into this myself. I'm not attached to `import.meta` in any way if there is a non hacky alternative we can use (I'd like to avoid the new Error() stacktrace hack if possible!)

**Comment by ingalls** [comment]
@jmroon I wonder if something like:

```js
const __dirname = __dirname || fileURLToPath(import.meta.url);
```

would work?

**Comment by ingalls** [comment]
@jmroon Have a proof of concept up here https://github.com/dfpc-coe/node-CoT/pull/30

**Comment by jmroon** [comment]
> @jmroon Have a proof of concept up here https://github.com/dfpc-coe/node-CoT/pull/30

I gave it a go. There were a few issues I was able to get around and got it up and running. I'm off for the next week but I'll be happy to put up a PR with the necessary changes!

The dirname issue actually ends up being even easier in practice. Tsup actually has a shim for it! Beyond that, it's mainly up to you how you'd like it distributed. Tsup could be used for both cjs/esm, and they can be delivered side by side (JS extension for ESM, CJS extension for CommonJS). Or they can be their own individual builds. 

I have a branch up but it's mainly the result of experimentation right now. I'm off on vacation next week but I'll try to get a PR the week of Sept 9!
